server/views/pages/request/visitRequestsListing.njk

```diff
@@ -0,0 +1,124 @@
+{% extends "layout.njk" %}
+{% from "partials/breadcrumbs.njk" import breadcrumbs with context %}
+{% from "govuk/components/table/macro.njk" import govukTable %}
+
+{% set pageHeaderTitle = "Requested visits" %}
+
+{% block beforeContent %}
+  {{ breadcrumbs() }}
+{% endblock %}
+
+{% set requestListRows = [] %}
+{% for visitRequest in visitRequests %}
```

Comment:
Just thinking, for clarity in the template, could this data not be created in the route then passed through?

Reply:
Yeah, might be worth a try 🤔. Just built rows in the template out of habit tbh as we've tended to do that - but it's not particularly pretty. I am using Nunjucks filters in places to build it up - but our own ones at least are just calls to `utils.ts` etc anyway so would work. Would need to think about escaping as you need to do this manually when passing an `html` value to a GOV.UK macro...

I'll merge this as it is for now but will try refactoring as suggested as a separate PR to see what it looks like 👍